java/test_flatbuf1/Test.java

```diff
@@ -16,16 +17,246 @@
+                break;
+            }
+
+            // output relation EO(e: bigint)
+            case flatbufTestRelation.EO: {
+                EOReader v = (EOReader)command.value();
+                System.out.println("From " + relid + " " + command.kind() + " EO{" + v.e() + "}");
+                break;
+            }
+            
+            // output relation FO(s: string)
+            case flatbufTestRelation.FO: {
+                FOReader v = (FOReader)command.value();
+                System.out.println("From " + relid + " " + command.kind() + " FO{\"" + v.s() + "\"}");
+                break;
+            }
+            
+            // output relation GO(d: bit<64>)
+            case flatbufTestRelation.GO: {
+                GOReader v = (GOReader)command.value();
+                System.out.println("From " + relid + " " + command.kind() + " GO{" + v.d() + "}");
+                break;
+            }
+            
+            // output relation HO(d: bit<128>)
+            case flatbufTestRelation.HO: {
+                HOReader v = (HOReader)command.value();
+                System.out.println("From " + relid + " " + command.kind() + " HO{" + v.d() + "}");
+                break;
+            }
+
+            // output relation IO(d: bit<12>)
+            case flatbufTestRelation.IO: {
+                IOReader v = (IOReader)command.value();
+                System.out.println("From " + relid + " " + command.kind() + " IO{" + v.d() + "}");
+                break;
+            }
+
+            // output relation JO(a: (bool, bit<8>, string))
+            case flatbufTestRelation.JO: {
+                JOReader v = (JOReader)command.value();
+                Tuple3__bool__bit_8___stringReader a = v.a();
+                System.out.println("From " + relid + " " + command.kind() + " JO{" + printTuple(a) + "}");
+                break;
+            }
+            
+            // output relation KO(t: tuple)
+            case flatbufTestRelation.KO: {
+                KOReader v = (KOReader)command.value();
+                Tuple3__bool__bit_8___stringReader t = v.t();
+                System.out.println("From " + relid + " " + command.kind() + " KO{" + printTuple(t) + "}");
+                break;
+            }
+
+            // output relation LO[tuple]
+            case flatbufTestRelation.LO: {
+                Tuple3__bool__bit_8___stringReader v = (Tuple3__bool__bit_8___stringReader)command.value();
+                System.out.println("From " + relid + " " + command.kind() + " " + printTuple(v));
+                break;
+            }
+
+            // output relation L0O(a: bool, b: bit<8>, s: string)
+            case flatbufTestRelation.L0O: {
+                L0OReader v = (L0OReader)command.value();
+                System.out.println("From " + relid + " " + command.kind() + " L0O{" + v.a() + ", " + v.b() + ", \"" + v.s() + "\"}");
```

Comment:
Is this correct if s has quotes inside? Or do you need to escape some characters?

Reply:
In fact, the same bug exists in DDlogRecord.java already.
We have to find out the rules for printing strings from Rust.

Reply:
I think this is fine for the purposes of this test, as we are not trying to generate something that parses as a DDlog or Java string literal. But yes, in `DDlogRecord.java` it is a bug. Also, strangely I couldn't find a standard library for printing strings with escaping in Java.